pytorch_lightning/accelerators/accelerator.py

```diff
@@ -175,6 +175,20 @@ def sync_tensor(self,
         """
         raise NotImplementedError()
 
+    def all_gather(self, tensor: Union[torch.Tensor], group: Optional[Any] = None, sync_grads: bool = False):
+        """
+        Function to gather a tensor from several distributed processes
+
+        Args:
+            tensor: tensor of shape (batch, ...)
+            group: the process group to gather results from. Defaults to all processes (world)
+            sync_grads: flag that allows users to synchronize gradients for all_gather op
+
+        Return:
+            A tensor of shape (world_size, batch, ...)
+        """
+        raise NotImplementedError()
```

Comment:
```suggestion
        return all_gather_ddp_if_available(tensor, group=group, sync_grads=sync_grads)
```
it seems to be everywhere the same, can we rather implement it here and drop in child if needed

Reply:
I would suggest against this, because we shouldn't assume that it can be directly implemented in every accelerator, a lot of duplication I know but it's best since theres no accelerator subclass

Reply:
@Borda @SeanNaren I agree with Sean, majority of accelerators are some variant of DDP, that is why there is quite a bit of duplication. However, the same thing won't work for TPU and horovod